Barkane/Assets/Scripts/Foldable Paper/PaperJoint.cs

```diff
@@ -10,13 +10,35 @@ public class PaperJoint : MonoBehaviour
     private bool isSelected = false; //true when this is the current selected fold
     public bool showLine = false; //true when this joint or any adjacent joins are selected. Used for showing visuals and partitioning graph
     public LineRenderer lineRenderer;
+    private PaperJoint currentJoint;
+    FoldablePaper foldablePaper;
+    List<GameObject> willBeFoldedAll;
+    private bool isFirstCall = true;
+    List<PaperSqaure> willBeFoldedPaperSquares = new List<PaperSqaure>();
+
     [SerializeField] private List<PaperJoint> adjList = new List<PaperJoint>();
 
     [SerializeField] private CapsuleCollider capsuleCollider;
    // private bool JointEnabled = true; //CO: Set to false to "cut" the paper along the given joint
     public bool canFold = true; //CO: Set to false to lock the current joint in position, as if the squares were glued together
 
-    private void Start() {
+    void Update(){
+        if (isSelected) {
+            if (isFirstCall) {
+                isFirstCall = !isFirstCall;
+                foldablePaper = FindObjectOfType<FoldablePaper>();
+                foldablePaper.FindFoldObjects();
+                willBeFoldedAll = foldablePaper.getFoldSide();
+                GetWillBeFoldedPaperSquares();
```

Comment:
Do all of these things need to be done in every joint? IE we only need to find the list of to be folded squares once, so can we update that component in foldable paper? This could also be dependent on changes to fold object list (making it a struct w square and joint lists) i might make soon, but I still think the GetWillBeFoldedPaperSquares should be moved to avoid duplication

Reply:
Moved the creating of the list of just the paper squares to fold to FoldablePaper. I can change it again if/when the changes to fold object list happen